druntime/src/object.d

```diff
@@ -2535,38 +2535,71 @@ class Throwable : Object
      * $(D Throwable) is thrown from inside a $(D catch) block. The originally
      * caught $(D Exception) will be chained to the new $(D Throwable) via this
      * field.
+     *
+     * If the 0 bit is set in this, it means the next exception is refcounted,
+     * meaning this object owns that object and should destroy it on
+     * destruction.
+     *
+     * WARNING: This means we are storing an interior pointer to the next in
+     * chain, to signify that the next in the chain is actually reference
+     * counted. We rely on the fact that a pointer to a Throwable is not 1-byte
+     * aligned. It is important to use a local field to indicate reference
+     * counting since we are not allowed to look at GC-allocated references
+     * inside the destructor. It is very important to store this as a void*,
+     * since optimizers can consider an unaligned pointer to not exist.
      */
-    private Throwable   nextInChain;
+    private void*   nextInChainPtr;
+
+    // 0 = not ref counted, 1 = ref counted
+    private @property size_t _nextIsRefcounted() @trusted scope pure nothrow @nogc const
```

Comment:
I think `bool` should work too and be usable as pointer offset, but not 100% sure.

And nit: underscore prefix for a private property, but none for `nextInChainPtr`? Seems unusual.

Reply:
original was `nextInChain` But yes, it doesn't matter what we call it. I just wanted to call it something different so it wasn't accidentally used elsewhere.

I used `size_t` because then I don't have to cast or remove the bits. But I think actually VRP will allow the `bool` return without masking off the other bits. I'll give it a try.